sdk/core/core-https/src/nodeHttpsClient.ts

```diff
@@ -158,7 +168,23 @@ export class NodeHttpsClient implements HttpsClient {
     } finally {
       // clean up event listener
       if (request.abortSignal && abortListener) {
-        request.abortSignal.removeEventListener("abort", abortListener);
+        let uploadStreamDone = Promise.resolve();
+        if (isReadableStream(body)) {
+          uploadStreamDone = isStreamComplete(body as NodeJS.ReadableStream);
+        }
+        let downloadStreamDone = Promise.resolve();
+        if (isReadableStream(responseStream)) {
+          downloadStreamDone = isStreamComplete(responseStream);
+        }
+
+        Promise.all([uploadStreamDone, downloadStreamDone])
+          .then(() => {
+            request.abortSignal?.removeEventListener("abort", abortListener!);
+            return;
+          })
+          .catch((e) => {
+            logger.warning("Error when cleaning up abortListener", e);
```

Comment:
Maybe a teensy more detail for folks who have to ponder the log message?

```suggestion
            logger.warning("Error when cleaning up abortListener on httpRequest", e);
```

Reply:
Certainly, updated.